### ISSUE TITLE
Error on https://www.disney--games.com/gameframe/1

### ISSUE THREAD

**Issue by paridhi16065** [ruffle-rs/ruffle#14338]
# Error Info
Error name: Error
Error message: panicked at /home/runner/.cargo/git/checkouts/nihav-vp6-660fee9efdd4622a/83c7e10/nihav-duck/src/codecs/vp56.rs:1030:18:
index out of bounds: the len is 2 but the index is 2
Error stack:
```
Error: panicked at /home/runner/.cargo/git/checkouts/nihav-vp6-660fee9efdd4622a/83c7e10/nihav-duck/src/codecs/vp56.rs:1030:18:
index out of bounds: the len is 2 but the index is 2
    at H.n.wbg.__wbg_new_a64e3f2afc2cf2f8 (https://unpkg.com/@ruffle-rs/ruffle/core.ruffle.62a8f0c82d1a76601817.js:1:56255)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::h2c4885bc68fd7d6d (wasm://wasm/03399e2a:wasm-function[9815]:0x9bac4c)
    at std::panicking::rust_panic_with_hook::hc20eadded6bfe687 (wasm://wasm/03399e2a:wasm-function[6301]:0x8acafe)
    at core::panicking::panic_fmt::h7a368385936888dc (wasm://wasm/03399e2a:wasm-function[9620]:0x9b5b68)
    at core::panicking::panic_bounds_check::ha1b736c04b755050 (wasm://wasm/03399e2a:wasm-function[7983]:0x933bd2)
    at nihav_duck::codecs::vp56::VP56Decoder::predict_dc::he097ff443be1aee4 (wasm://wasm/03399e2a:wasm-function[2222]:0x5e58de)
    at nihav_duck::codecs::vp56::VP56Decoder::decode_planes::h7dcc73d6830f38c7 (wasm://wasm/03399e2a:wasm-function[577]:0x839fc)
    at <ruffle_video_software::decoder::vp6::Vp6Decoder as ruffle_video_software::decoder::VideoDecoder>::decode_frame::he4824030118ae46d (wasm://wasm/03399e2a:wasm-function[776]:0x271c38)
    at <ruffle_video_software::backend::SoftwareVideoBackend as ruffle_video::backend::VideoBackend>::decode_video_stream_frame::hdb3a1b808c68f1f2 (wasm://wasm/03399e2a:wasm-function[2304]:0x5ffc4a)
    at ruffle_web::Ruffle::tick::heff5d35be58327d2 (wasm://wasm/03399e2a:wasm-function[858]:0x2ed883)
```

# Player Info
Allows script access: false
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "ANGLE (Intel, Intel(R) UHD Graphics (0x00009B41) Direct3D11 vs_5_0 ps_5_0, D3D11)"
Adapter Device Type: IntegratedGpu
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_COMPRESSION_BC | TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS | SHADER_UNUSED_VERTEX_OUTPUT)
Available features: Features(CLEAR_TEXTURE | MULTIVIEW)
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 1000, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 256, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: high
Surface samples: 4
Surface size: Extent3d { width: 750, height: 550, depth_or_array_layers: 1 }

# Page Info
Page URL: https://www.disney--games.com/gameframe/1
SWF URL: https://media.ngdfiles.com/hannah_montana/games/wireless-quest/wireless-quest.swf

# Browser Info
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 OPR/105.0.0.0
Platform: Win32
Has touch support: true

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-12-07
Channel: nightly
Built: 2023-12-07T00:15:04.300Z
Commit: 41ddb316a2b0a91c0d32e8f54eb0f020be42831f
Is extension: false

# Metadata
width: 580
height: 420
frameRate: 35
numFrames: 60
swfVersion: 8
backgroundColor: #31052F
isActionScript3: false
uncompressedLength: 3567270

**Comment by torokati44** [comment]
To reproduce:
 - Wait for Loading to finish
 - Click the "Play" button on the bottom left
 - Wait for the "Gotta hurry!" popup to go away, and a call to come in
 - If it's not a "video message" (which is random), then close and `GOTO 10`
 - If it is a video message, answer the call, and see a black frame instead of the video, then after a second or two, `panik

`cargo run 'https://media.ngdfiles.com/hannah_montana/games/wireless-quest/wireless-quest.swf'` works.

If you wish to download the SWF instead, make a folder named `Videos` next to it, and place these files in it:
 - https://media.ngdfiles.com/hannah_montana/games/wireless-quest/Videos/Hannah_Video_Trophy.flv
 - https://media.ngdfiles.com/hannah_montana/games/wireless-quest/Videos/Hannah_Video_Clock.flv
 - https://media.ngdfiles.com/hannah_montana/games/wireless-quest/Videos/Hannah_Video_Seattle.flv

**Comment by n0samu** [comment]
Fixed by #14339
